lib/yargs-parser.ts

```diff
@@ -243,7 +246,13 @@ export class YargsParser {
             // nargs format = '--f=monkey washing cat'
             i = eatNargs(i, m[1], args, m[2])
           } else {
-            setArg(m[1], m[2])
+            // only strip quotes if they wouldn't already be removed
+            if (shouldStripQuotes) {
+              setArg(m[1], m[2])
+            } else {
+              setArg(m[1], stripQuotes(m[2]))
```

Comment:
Since `setArgs` calls `processValue`, could this be simplified to:

```js
setArg(m[1],, m[2], shouldStripQuotes);
```

Reply:
shouldStripQuotes should always be forwarded by `setArg` to `processValue`, except at this line. In this invocation, shouldStripQuotes should be true. 

I'm not a fan of the solution I have [here](https://github.com/yargs/yargs-parser/blob/5ed073d480242be7c4959487ca12f39b4401e30a/lib/yargs-parser.ts#L546)

I try to keep code purely functional (where possible). In order to avoid reading global variables in setArg, I dynamically set the default value for `shouldStripQuotes`. I don't particularly like that pattern, so I'm down to change it.